Wikipedia/Code/WMFNewArticleTabsSettingsViewController.swift

```diff
@@ -0,0 +1,103 @@
+import UIKit
+import SwiftUI
+import WMFComponents
+import WMFData
+
+class WMFNewArticleTabsSettingsViewController: UIViewController {
+    private var hostingController: UIHostingController<WMFNewArticleTabSettingsView>?
+    private var viewModel: WMFNewArticleTabSettingsViewModel?
+    private let userDefaultsStore = WMFDataEnvironment.current.userDefaultsStore
+    let dataController = WMFArticleTabsDataController()
+    private var dataStore: MWKDataStore?
+    private var initialIndex: Int
+    private var theme: Theme
+    
+    @objc init(dataStore: MWKDataStore, theme: Theme) {
+        self.dataStore = dataStore
+        self.theme = theme
+        
+        let isBYREnabled = (try? userDefaultsStore?.load(key: WMFUserDefaultsKey.developerSettingsMoreDynamicTabsBYR.rawValue)) ?? false
+        let isDYKEnabled = (try? userDefaultsStore?.load(key: WMFUserDefaultsKey.developerSettingsMoreDynamicTabsDYK.rawValue)) ?? false
+
+        if isBYREnabled {
+            initialIndex = 0
+        } else if isDYKEnabled {
+            initialIndex = 1
+        } else {
+            initialIndex = 0
+        }
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    private func tagToKey(_ tag: Int) -> String {
+        switch tag {
+        case 0:
+            return WMFUserDefaultsKey.developerSettingsMoreDynamicTabsBYR.rawValue
+        case 1:
+            return WMFUserDefaultsKey.developerSettingsMoreDynamicTabsDYK.rawValue
+        default:
+            return ""
+        }
+    }
+    
+    private func saveSelection(selectedIndex: Int) {
+        let isBYR = selectedIndex == 0
+        let isDYK = selectedIndex == 1
+
+        try? userDefaultsStore?.save(key: WMFUserDefaultsKey.developerSettingsMoreDynamicTabsBYR.rawValue, value: isBYR)
+        try? userDefaultsStore?.save(key: WMFUserDefaultsKey.developerSettingsMoreDynamicTabsDYK.rawValue, value: isDYK)
+
+        dataController.moreDynamicTabsBYRIsEnabled = isBYR
+        dataController.moreDynamicTabsDYKIsEnabled = isDYK
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.viewModel = WMFNewArticleTabSettingsViewModel(
+            title: CommonStrings.tabsPreferencesTitle,
+            header: CommonStrings.newTabTheme,
+            options: [
+                CommonStrings.recommendations,
+                CommonStrings.didyouknow
+            ],
+            saveSelection: { [weak self] selectedIndex in
+                self?.saveSelection(selectedIndex: selectedIndex)
+            },
+            selectedIndex: initialIndex
+        )
+
+        guard let viewModel else { return }
+        
+        self.title = viewModel.title
+
+        let swiftUIView = WMFNewArticleTabSettingsView(viewModel: viewModel)
```

Comment:
(Optional) There's a WMFCanvasViewController that you may be able to subclass and save you some code here, it lets you embed a SwiftUI view the way you are doing. More of an FYI, this class is tiny as it is so just an optional suggestion.